src/react/elements.js

```diff
@@ -60,10 +71,10 @@ function deepTraverseAndFindOrDeleteFirstRenderProperties(
       let propValue = getProperty(realm, obj, propName);
 
       if (propName === "ref" || isEventProp(propName) || propValue instanceof FunctionValue) {
-        if (shouldDelete) {
+        if (shouldRemoveProps) {
           // We can do this, because we created the object as a fresh clone
           invariant(obj instanceof ObjectValue); // Make Flow happy
-          obj.properties.delete(propName);
+          canSafelyRemovePropertyDuringSerialization(realm, obj, propName);
```

Comment:
Maybe `markToSafely*` because `can*` looks like it's *asking* rather than setting